src/machine/machined-core.c

```diff
@@ -175,3 +181,220 @@ void manager_enqueue_gc(Manager *m) {
+                if (r == 0) {
+                        _cleanup_free_ struct local_address *addresses = NULL;
+                        struct local_address *a;
+                        int i, n;
+
+                        pair[0] = safe_close(pair[0]);
+
+                        n = local_addresses(NULL, 0, AF_UNSPEC, &addresses);
+                        if (n < 0)
+                                _exit(EXIT_FAILURE);
+
+                        for (a = addresses, i = 0; i < n; a++, i++) {
+                                struct iovec iov[2] = {
+                                        { .iov_base = &a->family, .iov_len = sizeof(a->family) },
+                                        { .iov_base = &a->address, .iov_len = FAMILY_ADDRESS_SIZE(a->family) },
+                                };
+
+                                r = writev(pair[1], iov, 2);
+                                if (r < 0)
+                                        _exit(EXIT_FAILURE);
+                        }
+
+                        pair[1] = safe_close(pair[1]);
+
+                        _exit(EXIT_SUCCESS);
+                }
+
+                pair[1] = safe_close(pair[1]);
+
+                _cleanup_free_ struct local_address *list = NULL;
+                size_t n_list = 0;
+
+                for (;;) {
+                        int family;
+                        ssize_t n;
+                        union in_addr_union in_addr;
+                        struct iovec iov[2];
+                        struct msghdr mh = {
+                                .msg_iov = iov,
+                                .msg_iovlen = 2,
+                        };
+
+                        iov[0] = IOVEC_MAKE(&family, sizeof(family));
+                        iov[1] = IOVEC_MAKE(&in_addr, sizeof(in_addr));
+
+                        n = recvmsg_safe(pair[0], &mh, 0);
+                        if (n < 0)
+                                return n;
+                        if ((size_t) n < sizeof(family))
+                                break;
+                        if ((size_t) n != sizeof(family) + FAMILY_ADDRESS_SIZE(family))
+                                return -EIO;
+
+                        r = add_local_address(&list,
+                                              &n_list,
+                                              /* ifindex = */ 0,
+                                              /* scope = */ '\0',
+                                              family,
+                                              &in_addr);
+                        if (r < 0)
+                                return r;
+                }
+
+                r = wait_for_terminate_and_check("(sd-addrns)", child, 0);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to wait for child: %m");
+                if (r != EXIT_SUCCESS) {
+                        log_debug("Child died abnormally, exit code=%d", r);
+                        return -ESHUTDOWN;
```

Comment:
Missed one here

Reply:
ESHUTDOWN is an artificial error code. It doesn't make sense to log it.